component/common/src/main/java/org/wso2/carbon/extension/identity/emailotp/common/EmailOtpServiceImpl.java

```diff
@@ -333,6 +353,25 @@ private ValidationResponseDTO isValid(SessionDTO sessionDTO, String emailOtp, St
             throws EmailOtpException {
 
         FailureReasonDTO error;
+        User user = getUserById(userId);
+        // Check if user account is locked?
+        if (checkAccountLock) {
+
+            if (Utils.isAccountLocked(user)) {
+                return createAccountLockedResponse(userId, showFailureReason);
+            }
+        }
+
+        if (Utils.isUserDisabled(user)) {
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("User account is disabled for the user : %s.", userId));
+            }
+            error = showFailureReason
```

Comment:
move this to a separate private method similar to locking scenatio

Reply:
fixed with e31be4c77fdedf379c272da521b6e2ab837b2a11